### PR TITLE
Introduce quarkus-rest-client-mutiny extension

### LINKED ISSUE #9709: Introduce a quarkus-rest-client-mutiny extension
**Describe the extension**
The existing `quarkus-resteasy-mutiny` extension (which is also mentioned in the https://quarkus.io/guides/rest-client guide) depends on the `quarkus-resteasy` extensions, which may not be adequate for applications that do not expose any endpoints (eg. command line apps) 

Interested in this extension, please +1 via the emoji/reaction feature of GitHub (top right).

### PR DESCRIPTION
**Pull request by mkouba**
- also introduce quarkus-resteasy-mutiny-common extension
- resolves #9709

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]
LGTM! Awesome!

File: extensions/rest-client-mutiny/runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+name: "REST Client Mutiny"
```
Comment (cescoffier) [inline extensions/rest-client-mutiny/runtime/src/main/resources/META-INF/quarkus-extension.yaml:2]: "Mutiny support for REST Client" maybe ?

**Review by cescoffier — APPROVED** [review]
We may have to update documentation and quickstart to use the new extension or at least to mention it.

**Comment by gsmet** [comment]
Please rebase when you fix the extension name (CI fix will come with it).

**Comment by mkouba** [comment]
Name fixed and rebased...

**Review by cescoffier — APPROVED** [review]

**Comment by jaikiran** [comment]
Failures look related:

```
2020-09-03T17:04:33.2968989Z Caused by: java.lang.RuntimeException: Failed to start quarkus
2020-09-03T17:04:33.2970424Z 	at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:403)
2020-09-03T17:04:33.2970779Z 	... 44 more
2020-09-03T17:04:33.2973272Z Caused by: java.lang.RuntimeException: Unable to find class for provider io.quarkus.resteasy.mutiny.common.runtime.UniInvokerProvider
2020-09-03T17:04:33.2975831Z 	at io.quarkus.restclient.runtime.RestClientRecorder.registerProviders(RestClientRecorder.java:84)
2020-09-03T17:04:33.2976843Z 	at io.quarkus.restclient.runtime.RestClientRecorder.initializeResteasyProviderFactory(RestClientRecorder.java:64)
2020-09-03T17:04:33.2979950Z 	at io.quarkus.deployment.steps.RestClientProcessor$registerProviders350329842.deploy_0(RestClientProcessor$registerProviders350329842.zig:668)
2020-09-03T17:04:33.2983273Z 	at io.quarkus.deployment.steps.RestClientProcessor$registerProviders350329842.deploy(RestClientProcessor$registerProviders350329842.zig:40)
2020-09-03T17:04:33.2983932Z 	at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:350)
2020-09-03T17:04:33.2984293Z 	... 44 more
2020-09-03T17:04:33.2986679Z Caused by: java.lang.ClassNotFoundException: io.quarkus.resteasy.mutiny.common.runtime.UniInvokerProvider
2020-09-03T17:04:33.2995551Z 	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:602)
2020-09-03T17:04:33.2997981Z 	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
2020-09-03T17:04:33.2998366Z 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
2020-09-03T17:04:33.2998718Z 	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:406)
2020-09-03T17:04:33.2999069Z 	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:363)
2020-09-03T17:04:33.2999432Z 	at io.quarkus.restclient.runtime.RestClientRecorder.registerProviders(RestClientRecorder.java:81)
2020-09-03T17:04:33.2999772Z 	... 48 more
```

**Comment by mkouba** [comment]
@jaikiran Yep, I'll take a look in a moment.

**Pull request merged**